src/Client.php

```diff
@@ -286,6 +286,29 @@ public function validateTokenClaims(array $tokenClaims): void
         }
     }
 
+    /**
+     * Requests and returns the resource owner of given access token.
+     *
+     * @throws InvalidTokenException when the ID token is invalid.
+     */
+    public function getResourceOwner(AccessTokenInterface $token): ResourceOwnerInterface
+    {
+        // If the ID token is not null, use to build the resource owner.
+        // Otherwise, use the claims from the access token.
+        if ($idToken = $token->getIdToken()) {
+            $tokenClaims = $this->decodeToken($idToken);
+        } else {
+            $tokenClaims = $this->decodeToken($token->getToken());
+        }
+
+        return $this->createResourceOwner($tokenClaims, $token);
+    }
+
+    protected function createResourceOwner(array $claims, AccessTokenInterface $token): ResourceOwnerInterface
```

Comment:
Passing `$token` in but it's never used?

Reply:
Per the `league/oauth-client` equivalent - the aim was to be consistent.